AppController/djinn.rb

```diff
@@ -1273,26 +1276,21 @@ def get_cluster_stats_json(secret)
   # Updates our locally cached information about the CPU, memory, and disk
   # usage of each machine in this AppScale deployment.
   def update_node_info_cache()
-    new_stats = []
-
     Thread.new {
-      @nodes.each { |node|
-        ip = node.private_ip
-        if ip == my_node.private_ip
-          node_stats = JSON.load(get_node_stats_json(@@secret))
-        else
-          acc = AppControllerClient.new(ip, @@secret)
-          begin
-            node_stats = JSON.load(acc.get_node_stats_json())
-          rescue FailedNodeException
-            Djinn.log_warn("Failed to get status update from node at #{ip}, so " +
-              "not adding it to our cached info.")
-            next
-          end
-        end
-        new_stats << node_stats
+      ip = get_shadow.private_ip
+      uri = URI("http://#{ip}:#{HermesService.getport()}/cluster_stats")
+      string_stats = (Net::HTTP.post_form(uri, 'secret' => @@secret))
```

Comment:
@cdonati I had been using a get but then it complained that the response body should be empty (if I remember correctly). I would be happy to change back to get because I felt like that was the correct thing to use as well.